media/js/firefox/new/all-test-variations-pt2.js

```diff
@@ -0,0 +1,127 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+(function($, dataLayer) {
+    'use strict';
+
+    // check page version for tests
+    var version = Number($('#masthead').data('version'));
+
+    // adds sort value & (optional) new text to direct download li
+    var prepLi = function(li) {
+        var $li = $(li);
+        var $liA = $li.find('a');
+        var newOsName;
+
+        switch ($liA.text()) {
+        case 'Windows':
+            newOsName = 'Windows 32-bit';
+            $li.attr('data-sort', 1);
+            break;
+        case 'Linux':
+            newOsName = 'Linux 32-bit';
+            $li.attr('data-sort', 2);
+            break;
+        case 'Windows 64-bit':
+            $li.attr('data-sort', 3);
+            break;
+        case 'Linux 64-bit':
+            $li.attr('data-sort', 4);
+            break;
+        case 'OS X':
+            $li.attr('data-sort', 5);
+            break;
+        default:
+            $li.attr('data-sort', 99);
+            break;
+        }
+
+        if (newOsName) {
+            $liA.text(newOsName);
+        }
+
+        return li;
+    };
+
+    var setupTest = function(version) {
+        var $dlButton = $('#download-button-desktop-release');
+        var $html = $('html');
+
+        // make sure desktop download button exists, user is not on Android or iOS, and user is on a recognized platform
+        if (!$html.hasClass('ios') && !$html.hasClass('android') && $dlButton.length && $dlButton.find('.unrecognized-download:visible').length === 0) {
+            // pull the nojs links out of the primary CTA
+            var $directLis = $dlButton.find('.nojs-download li').remove();
```

Comment:
Agreed, but I'm holding off on that until we have a final direction. I really hope we go with a single link to an improved `/firefox/all/` rather than injecting direct links here.